tiledb/sm/query/writers/writer_base.cc

```diff
@@ -468,6 +468,112 @@ void WriterBase::check_subarray() const {
+  switch (type) {
+    case Datatype::INT8:
+      return increasing ?
+                 buffer.is_sorted<int8_t, std::less_equal<int8_t>>() :
+                 buffer.is_sorted<int8_t, std::greater_equal<int8_t>>();
+    case Datatype::UINT8:
+      return increasing ?
+                 buffer.is_sorted<uint8_t, std::less_equal<uint8_t>>() :
+                 buffer.is_sorted<uint8_t, std::greater_equal<uint8_t>>();
+    case Datatype::INT16:
+      return increasing ?
+                 buffer.is_sorted<int16_t, std::less_equal<int16_t>>() :
+                 buffer.is_sorted<int16_t, std::greater_equal<int16_t>>();
+    case Datatype::UINT16:
+      return increasing ?
+                 buffer.is_sorted<uint16_t, std::less_equal<uint16_t>>() :
+                 buffer.is_sorted<uint16_t, std::greater_equal<uint16_t>>();
+    case Datatype::INT32:
+      return increasing ?
+                 buffer.is_sorted<int32_t, std::less_equal<int32_t>>() :
+                 buffer.is_sorted<int32_t, std::greater_equal<int32_t>>();
+    case Datatype::UINT32:
+      return increasing ?
+                 buffer.is_sorted<uint32_t, std::less_equal<uint32_t>>() :
+                 buffer.is_sorted<uint32_t, std::greater_equal<uint32_t>>();
+    case Datatype::INT64:
+      return increasing ?
+                 buffer.is_sorted<int64_t, std::less_equal<int64_t>>() :
+                 buffer.is_sorted<int64_t, std::greater_equal<int64_t>>();
+    case Datatype::UINT64:
+      return increasing ?
+                 buffer.is_sorted<uint64_t, std::less_equal<uint64_t>>() :
+                 buffer.is_sorted<uint64_t, std::greater_equal<uint64_t>>();
+    case Datatype::FLOAT32:
+      return increasing ? buffer.is_sorted<float, std::less_equal<float>>() :
+                          buffer.is_sorted<float, std::greater_equal<float>>();
+    case Datatype::FLOAT64:
+      return increasing ?
+                 buffer.is_sorted<double, std::less_equal<double>>() :
+                 buffer.is_sorted<double, std::greater_equal<double>>();
+    case Datatype::DATETIME_YEAR:
+    case Datatype::DATETIME_MONTH:
+    case Datatype::DATETIME_WEEK:
+    case Datatype::DATETIME_DAY:
+    case Datatype::DATETIME_HR:
+    case Datatype::DATETIME_MIN:
+    case Datatype::DATETIME_SEC:
+    case Datatype::DATETIME_MS:
+    case Datatype::DATETIME_US:
+    case Datatype::DATETIME_NS:
+    case Datatype::DATETIME_PS:
+    case Datatype::DATETIME_FS:
+    case Datatype::DATETIME_AS:
+    case Datatype::TIME_HR:
+    case Datatype::TIME_MIN:
+    case Datatype::TIME_SEC:
+    case Datatype::TIME_MS:
+    case Datatype::TIME_US:
+    case Datatype::TIME_NS:
+    case Datatype::TIME_PS:
+    case Datatype::TIME_FS:
+    case Datatype::TIME_AS:
+      return increasing ?
+                 buffer.is_sorted<int64_t, std::less_equal<int64_t>>() :
+                 buffer.is_sorted<int64_t, std::greater_equal<int64_t>>();
+    case Datatype::STRING_ASCII:
```

Comment:
There are fixed size strings. I don't think this will work for those?

Reply:
As discussed, currently fixed-length ordered attributes are not allowed, but we should implement that feature in a follow-up.